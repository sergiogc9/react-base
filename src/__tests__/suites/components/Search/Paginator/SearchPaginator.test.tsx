import { mount } from "enzyme";
import React from "react";
import SearchPaginator from "@src/components/Search/Paginator/SearchPaginator";
import GA from '@src/lib/googleAnalytics';

let onSetStartMock: any;

describe("Search Form", () => {
	let wrapper: any;

	beforeEach(() => {
		onSetStartMock = jest.fn();
	});

	it("Test component snapshot loading results", () => {
		wrapper = mount(
			<SearchPaginator
				limit={20}
				start={0}
				total={0}
				onSetStart={onSetStartMock}
				loadingDocuments={true}
			/>
		);
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot loading results without results", () => {
		wrapper = mount(
			<SearchPaginator
				limit={20}
				start={0}
				total={0}
				onSetStart={onSetStartMock}
				loadingDocuments={false}
			/>
		);
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot 5000 results", () => {
		wrapper = mount(
			<SearchPaginator
				limit={20}
				start={0}
				total={5000}
				onSetStart={onSetStartMock}
				loadingDocuments={false}
			/>
		);
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot > 5000 results", () => {
		wrapper = mount(
			<SearchPaginator
				limit={20}
				start={0}
				total={6000}
				onSetStart={onSetStartMock}
				loadingDocuments={false}
			/>
		);
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test change page", () => {
		GA.trackEvent = jest.fn();
		wrapper = mount(
			<SearchPaginator
				limit={20}
				start={0}
				total={5000}
				onSetStart={onSetStartMock}
				loadingDocuments={false}
			/>
		);
		expect(GA.trackEvent).toHaveBeenCalledTimes(0);
		expect(onSetStartMock).toHaveBeenCalledTimes(0);
		wrapper.find("#paginator button").at(0).simulate('click');
		expect(GA.trackEvent).toHaveBeenCalledTimes(0);
		expect(onSetStartMock).toHaveBeenCalledTimes(0);
		wrapper.find("#paginator button").at(1).simulate('click');
		expect(GA.trackEvent).toHaveBeenCalledTimes(1);
		expect(onSetStartMock).toHaveBeenCalledTimes(1);
	});

	it("Test change prev page", () => {
		wrapper = mount(
			<SearchPaginator
				limit={20}
				start={20}
				total={5000}
				onSetStart={onSetStartMock}
				loadingDocuments={false}
			/>
		);
		expect(onSetStartMock).toHaveBeenCalledTimes(0);
		wrapper.find("#paginator button").at(0).simulate('click');
		expect(onSetStartMock).toHaveBeenCalledTimes(1);
	});

	it("Test change next page", () => {
		wrapper = mount(
			<SearchPaginator
				limit={20}
				start={0}
				total={5000}
				onSetStart={onSetStartMock}
				loadingDocuments={false}
			/>
		);
		expect(onSetStartMock).toHaveBeenCalledTimes(0);
		wrapper.find("#paginator button").at(6).simulate('click');
		expect(onSetStartMock).toHaveBeenCalledTimes(1);
	});

});
