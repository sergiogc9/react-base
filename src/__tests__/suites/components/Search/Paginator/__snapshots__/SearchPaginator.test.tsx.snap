// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form Test component snapshot > 5000 results 1`] = `"<div id=\\"paginator\\"><button type=\\"button\\" disabled=\\"\\" class=\\"md-btn md-btn--flat md-btn--text md-text--disabled md-inline-block selected\\">1</button><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div>2</button><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div>3</button><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div>4</button><button type=\\"button\\" disabled=\\"\\" class=\\"md-btn md-btn--flat md-btn--text md-text--disabled md-inline-block\\">...</button><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div>250</button><button title=\\"resuts.pagination.next_page\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div> &gt;&gt; </button></div>"`;

exports[`Search Form Test component snapshot 5000 results 1`] = `"<div id=\\"paginator\\"><button type=\\"button\\" disabled=\\"\\" class=\\"md-btn md-btn--flat md-btn--text md-text--disabled md-inline-block selected\\">1</button><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div>2</button><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div>3</button><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div>4</button><button type=\\"button\\" disabled=\\"\\" class=\\"md-btn md-btn--flat md-btn--text md-text--disabled md-inline-block\\">...</button><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div>250</button><button title=\\"resuts.pagination.next_page\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div> &gt;&gt; </button></div>"`;

exports[`Search Form Test component snapshot loading results 1`] = `null`;

exports[`Search Form Test component snapshot loading results without results 1`] = `null`;
